File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let mp = 0;
    let bought_price = prices[0];

    for (const today_price of prices.slice(1)) {
        if (bought_price > today_price) {
            bought_price = today_price;
        }
        mp = Math.max(mp, today_price - bought_price);
    }
    return mp;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mp = 0
        bought_price = prices[0]

        for today_price in prices[1:]:
            if bought_price > today_price:
                bought_price = today_price
            
            mp = max(mp, today_price - bought_price)
        
        return mp